packages/automation/README.md

```diff
@@ -0,0 +1,347 @@
+  stateMachine.addTransition({
+    // Because this transition does not have any listeners, it will be triggered automatically when the machine enters fromState
+    fromState: 'autoAdvancingState',
+    toState: 'listenBlocks',
+  });
+  stateMachine.addTransition({
+    fromState: 'listenBlocks',
+    toState: 'autoAdvancingState',
+    // listeners are the ones that will produce the values that the transition will monitor
+    listeners: [new EVMBlockListener(LIT_EVM_CHAINS.ethereum.rpcUrls[0])],
+    // check is the function that will evaluate all values produced by listeners and define if there is a match or not
+    check: async (values): Promise<boolean> => {
+      // values are the results of all listeners
+      const blockData = values[0] as BlockData;
+      if (!blockData) return false;
+      console.log(`New block: ${blockData.number} (${blockData.hash})`);
+      return blockData.hash.endsWith('0');
+    },
+    // when check finds a match (returns true) this function gets executed and the machine moves to toState
+    onMatch: async (values) => {
+      // values are the results of all listeners
+      console.log('We have matching values here');
+    },
+    onMismatch: undefined, // when check returns false (there is a mismatch) this function gets executed but the machine does not change state
+    onError: undefined,
+  });
+
+  await stateMachine.startMachine('listenBlocks');
+}
+monitorEthereumBlocksWithHashEndingWithZero().catch(console.error);
+```
+
+Last machine could have been implemented with just the `listenBlocks` state and a `listenBlocks` -> `listenBlocks` transition, but the machine realizes that the state does not change and therefore does not exit nor enter the state, however it runs the transition `onMatch` function.
+
+## Context
+
+Each State Machine has its own information repository called `context`.
+
+When using the defined states in the declarative interface, some values are already populated and then used later
+
+- `StateDefinition.usePkp` populates `context.activePkp` with the minted PKP data
+- `StateDefinition.useCapacityNFT` populates `context.activeCapacityTokenId` with the minted Capacity Token Id
+- `StateDefinition.litAction` populates `context.lastLitActionResponse` with the lit action response
+- `StateDefinition.transaction` populates `context.lastTransactionReceipt` with the transaction receipt
+
+Several places in the machine definition can read values from the context. Instead of passing a literal value, pass an object with the `contextPath` property, like in the following example.
+
+The machine context can be accessed using its `getFromContext`, `setToContext` or `pushToContext` methods to read or write.
+
+### Advance example
+
+By leveraging the State Machine context and the ability of Lit PKPs to sign transaction of a variety of chains, it is possible to implement a Token Bridge that composes multiple chains and even offchain interaction if needed among other uses cases.
```

Comment:
a bit hesitant to use 'token bridge' here because of the connotations associated with it. Instead, maybe we can say 

"By leveraging context from the State Machine in combination with Lit PKPs, it is possible to implement a cross-chain messaging service that can be used to read and write data across virtually any blockchain"

Reply:
Good point. Changed